src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/MemoryAddressImpl.java

```diff
@@ -267,110 +270,216 @@ public void set(ValueLayout.OfAddress layout, long offset, Addressable value) {
     @CallerSensitive
     public char getAtIndex(ValueLayout.OfChar layout, long index) {
         Reflection.ensureNativeAccess(Reflection.getCallerClass());
-        return NativeMemorySegmentImpl.EVERYTHING.get(layout, toRawLongValue() + (index * layout.byteSize()));
+        if (layout.byteAlignment() <= layout.byteSize()) {
```

Comment:
Here we also have to workaround [1]. Since we don't want to slice the everything segment (as that would create an additional instance), the only option is to use the var handle directly, but to use the "aligned" version if we detect that the offset is always aligned. One caveat is that we need to manually check the base address for alignment, since this check will be omitted otherwise. Again, this logic will go away when [1] is fixed.

[1] - https://bugs.openjdk.java.net/browse/JDK-8277850